### PR TITLE
Clean up grammar source

### PR DESCRIPTION
**Pull request by Alhadis**
WYSIWYG.

The diffs will tell you more than my tired babble will.

[Here's a list](http://www.schweikhardt.net/identifiers.html) to provide keyword copy+pasta if it helps accelerate review. =)

### CODE REVIEW

File: grammars/c++.cson [hunk 1]
```diff
@@ -174,7 +198,15 @@
         ]
       }
       {
-        'begin': '(?x)\n    \t\t\t\t(:)                            # begin-of-line\n    \t\t\t\t((?=\\s*[A-Za-z_][A-Za-z0-9_:]* # actual name\n    \t\t\t\t \\s*(\\()))                      # start bracket or end-of-line\n    \t\t\t'
+        'begin': '''(?x)
+          (:) # beginning of line
```
Comment (50Wliu) [inline grammars/c++.cson]: Incorrect comment

File: grammars/c++.cson [hunk 2]
```diff
@@ -98,7 +105,14 @@
...
   }
   {
-    'begin': '(?x)\n    \t\t\t\t(?:  ^                                 # begin-of-line\n    \t\t\t\t  |  (?: (?<!else|new|=) )             #  or word + space before name\n    \t\t\t\t)\n    \t\t\t\t((?:[A-Za-z_][A-Za-z0-9_]*::)*+~[A-Za-z_][A-Za-z0-9_]*) # actual name\n    \t\t\t\t \\s*(\\()                           # terminating semi-colon\n    \t\t\t'
+    'begin': '''(?x)
+      (?:
+        ^ |                  # beginning of line
+        (?:(?<!else|new|=))  # or word + space before name
+      )
+      ((?:[A-Za-z_][A-Za-z0-9_]*::)*+~[A-Za-z_][A-Za-z0-9_]*) # actual name
+      \\s*(\\()              # terminating semicolon
```
Comment (50Wliu) [inline grammars/c++.cson]: incorrect comment

File: grammars/c++.cson [hunk 3]
```diff
@@ -80,7 +80,14 @@
...
   }
   {
-    'begin': '(?x)\n    \t\t\t\t(?:  ^                                 # begin-of-line\n    \t\t\t\t  |  (?: (?<!else|new|=) )             #  or word + space before name\n    \t\t\t\t)\n    \t\t\t\t((?:[A-Za-z_][A-Za-z0-9_]*::)*+~[A-Za-z_][A-Za-z0-9_]*) # actual name\n    \t\t\t\t \\s*(\\()                           # start bracket or end-of-line\n    \t\t\t'
+    'begin': '''(?x)
+      (?:
+        ^ |                  # beginning of line
+        (?:(?<!else|new|=))  # or word + space before name
+      )
+      ((?:[A-Za-z_][A-Za-z0-9_]*::)*+~[A-Za-z_][A-Za-z0-9_]*) # actual name
+      \\s*(\\()              # start bracket or end-of-line
```
Comment (50Wliu) [inline grammars/c++.cson]: bracket -> parentheses
Also, there is no end-of-line match here

File: grammars/c++.cson [hunk 4]
```diff
@@ -146,7 +160,13 @@
             'name': 'support.function.any-method.c'
           '2':
             'name': 'punctuation.definition.parameters.c'
-        'match': '(?x)\n    \t\t\t\t(\n    \t\t\t\t\t(?!while|for|do|if|else|switch|catch|enumerate|return|r?iterate)(?: \\b[A-Za-z_][A-Za-z0-9_]*+\\b | :: )*+                  # actual name\n    \t\t\t\t)\n    \t\t\t\t \\s*(\\()'
+        'match': '''(?x)
+          (
+            (?!while|for|do|if|else|switch|catch|enumerate|return|r?iterate)
+            (?:\\b[A-Za-z_][A-Za-z0-9_]*+\\b|::)*+ # actual name
+          )
+          \\s*(\\()
```
Comment (50Wliu) [inline grammars/c++.cson]: missing comment

File: grammars/c++.cson [hunk 5]
```diff
@@ -156,7 +176,11 @@
   'constructor':
     'patterns': [
       {
-        'begin': '(?x)\n    \t\t\t\t(?:  ^\\s*)                             # begin-of-line\n    \t\t\t\t((?!while|for|do|if|else|switch|catch|enumerate|r?iterate)[A-Za-z_][A-Za-z0-9_:]*) # actual name\n    \t\t\t\t \\s*(\\()                            # start bracket or end-of-line\n    \t\t\t'
+        'begin': '''(?x)
+          (?:^\\s*)  # beginning of line
+          ((?!while|for|do|if|else|switch|catch|enumerate|r?iterate)[A-Za-z_][A-Za-z0-9_:]*) # actual name
+          \\s*(\\()  # start bracket or end-of-line
```
Comment (50Wliu) [inline grammars/c++.cson]: bracket -> parentheses
Also, there is no end-of-line match here

File: grammars/c++.cson [hunk 6]
```diff
@@ -174,7 +198,15 @@
         ]
       }
       {
-        'begin': '(?x)\n    \t\t\t\t(:)                            # begin-of-line\n    \t\t\t\t((?=\\s*[A-Za-z_][A-Za-z0-9_:]* # actual name\n    \t\t\t\t \\s*(\\()))                      # start bracket or end-of-line\n    \t\t\t'
+        'begin': '''(?x)
+          (:) # beginning of line
+          (
+            (?=
+              \\s*[A-Za-z_][A-Za-z0-9_:]* # actual name
+              \\s* (\\() # start bracket or end-of-line
```
Comment (50Wliu) [inline grammars/c++.cson]: bracket -> parentheses
Also, there is no end-of-line match here

File: grammars/c.cson [hunk 7]
```diff
@@ -196,19 +196,33 @@
...
   }
   {
-    'match': '\\b(int8_t|int16_t|int32_t|int64_t|uint8_t|uint16_t|uint32_t|uint64_t|int_least8_t|int_least16_t|int_least32_t|int_least64_t|uint_least8_t|uint_least16_t|uint_least32_t|uint_least64_t|int_fast8_t|int_fast16_t|int_fast32_t|int_fast64_t|uint_fast8_t|uint_fast16_t|uint_fast32_t|uint_fast64_t|intptr_t|uintptr_t|intmax_t|intmax_t|uintmax_t|uintmax_t)\\b'
+    'match': '''(?x) \\b
+      (int8_t|int16_t|int32_t|int64_t|uint8_t|uint16_t|uint32_t|uint64_t|int_least8_t
+      |int_least16_t|int_least32_t|int_least64_t|uint_least8_t|uint_least16_t|uint_least32_t
+      |uint_least64_t|int_fast8_t|int_fast16_t|int_fast32_t|int_fast64_t|uint_fast8_t
+      |uint_fast16_t|uint_fast32_t|uint_fast64_t|intptr_t|uintptr_t|intmax_t|intmax_t
+      |uintmax_t|uintmax_t)
+    \\b'''
```
Comment (50Wliu) [inline grammars/c.cson]: Should be indented one more

File: grammars/c.cson [hunk 8]
```diff
@@ -196,19 +196,33 @@
...
+    'match': '''(?x) \\b
+      (AbsoluteTime|Boolean|Byte|ByteCount|ByteOffset|BytePtr|CompTimeValue|ConstLogicalAddress|ConstStrFileNameParam
+      |ConstStringPtr|Duration|Fixed|FixedPtr|Float32|Float32Point|Float64|Float80|Float96|FourCharCode|Fract|FractPtr
+      |Handle|ItemCount|LogicalAddress|OptionBits|OSErr|OSStatus|OSType|OSTypePtr|PhysicalAddress|ProcessSerialNumber
+      |ProcessSerialNumberPtr|ProcHandle|Ptr|ResType|ResTypePtr|ShortFixed|ShortFixedPtr|SignedByte|SInt16|SInt32|SInt64
+      |SInt8|Size|StrFileName|StringHandle|StringPtr|TimeBase|TimeRecord|TimeScale|TimeValue|TimeValue64|UInt16|UInt32
+      |UInt64|UInt8|UniChar|UniCharCount|UniCharCountPtr|UniCharPtr|UnicodeScalarValue|UniversalProcHandle|UniversalProcPtr
+      |UnsignedFixed|UnsignedFixedPtr|UnsignedWide|UTF16Char|UTF32Char|UTF8Char)
+    \\b'''
```
Comment (50Wliu) [inline grammars/c.cson]: Should be indented one more

File: grammars/c.cson [hunk 9]
```diff
@@ -320,7 +334,20 @@
...
+              (?=\\s)(?<!else|new|return)
+              (?<=\\w) \\s+  # or word + space before name
+            )
+          )
+          (
+            (?:[A-Za-z_][A-Za-z0-9_]*+ | :: )++   # actual name
+            |
+            (?:(?<=operator) (?:[-*&<>=+!]+ | \\(\\) | \\[\\]))  # if it is a C++ operator
+          )
+          \\s*(\\()
```
Comment (50Wliu) [inline grammars/c.cson]: missing comment

File: grammars/c.cson [hunk 10]
```diff
@@ -320,7 +334,20 @@
...
+          (?:
+            (?:
+              (?=\\s)(?<!else|new|return)
+              (?<=\\w) \\s+  # or word + space before name
+            )
+          )
+          (
+            (?:[A-Za-z_][A-Za-z0-9_]*+ | :: )++   # actual name
+            |
+            (?:(?<=operator) (?:[-*&<>=+!]+ | \\(\\) | \\[\\]))  # if it is a C++ operator
```
Comment (50Wliu) [inline grammars/c.cson]: this is `c.cson` though...

File: grammars/c.cson [hunk 11]
```diff
@@ -338,7 +365,22 @@
...
+      
+      # Actual name
+      (
+        \\b (?!
+          (while|for|do|if|else|switch|catch|enumerate|return|sizeof|[cr]?iterate)
+          \\s* \\(
+        )
+        (?:(?!NS)[A-Za-z_][A-Za-z0-9_]*+\\b | ::)++
+      )
+      \\s*(\\()
```
Comment (50Wliu) [inline grammars/c.cson]: missing comment

File: grammars/c.cson [hunk 12]
```diff
@@ -338,7 +365,22 @@
...
+    'match': '''(?x)
+      (?:
+        (?=\\s)
+        (?:(?<=else|new|return) | (?<!\\w)) (\\s+)
+      )?
+      
+      # Actual name
+      (
+        \\b (?!
+          (while|for|do|if|else|switch|catch|enumerate|return|sizeof|[cr]?iterate)
```
Comment (50Wliu) [inline grammars/c.cson]: I believe only riterate exists.

File: grammars/c.cson [hunk 13]
```diff
@@ -338,7 +365,22 @@
...
-    'match': '(?x) (?: (?= \\s )  (?:(?<=else|new|return) | (?<!\\w)) (\\s+))?\n\t\t\t(\\b \n\t\t\t\t(?!(while|for|do|if|else|switch|catch|enumerate|return|sizeof|[cr]?iterate)\\s*\\()(?:(?!NS)[A-Za-z_][A-Za-z0-9_]*+\\b | :: )++                  # actual name\n\t\t\t)\n\t\t\t \\s*(\\()'
+    'match': '''(?x)
+      (?:
+        (?=\\s)
+        (?:(?<=else|new|return) | (?<!\\w)) (\\s+)
+      )?
+      
+      # Actual name
+      (
+        \\b (?!
```
Comment (50Wliu) [inline grammars/c.cson]: This `(?!` doesn't match up with the closing parentheses.  Maybe place it on a new line so it does?

File: grammars/c.cson [hunk 14]
```diff
@@ -338,7 +365,22 @@
...
+          (while|for|do|if|else|switch|catch|enumerate|return|sizeof|[cr]?iterate)
+          \\s* \\(
+        )
+        (?:(?!NS)[A-Za-z_][A-Za-z0-9_]*+\\b | ::)++
```
Comment (50Wliu) [inline grammars/c.cson]: Maybe add a comment for this?

**Comment by 50Wliu** [comment]
Some (most) of my comments aren't directly related to this change since you just reorganized everything and kept the existing comments.  Just some general things that caught my eye though.

**Comment by Alhadis** [comment]
Can I nuke the comments altogether? They don't appear to be adding anything of value.

**Comment by 50Wliu** [comment]
I do prefer the comments as they succinctly tell you what each line is supposed to do.

File: grammars/c++.cson (hunk 6 above)
Comment (Alhadis) [inline grammars/c++.cson]: A parenthesis *is* a bracket.

File: grammars/c.cson (hunk 10 above)
Comment (Alhadis) [inline grammars/c.cson]: I'll interpret that as "delete this".

**Pull request merged**